File: 1266. Minimum Time Visiting All Points/code.py
from typing import List

class Solution:
    def minTimeToVisitAllPoints(self, points: List[List[int]]) -> int:
        time = 0
        current = points.pop(0)
        for point in points:
            while True:
                if current == point:
                    break
                if current[0] < point [0]:
                    current[0] += 1
                elif current[0] > point [0]:
                    current[0] -= 1
                if current[1] < point [1]:
                    current[1] += 1
                elif current[1] > point [1]:
                    current[1] -= 1
                time += 1
        return time
    
points = [[1,1],[3,4],[-1,0]]

sol = Solution()
print(sol.minTimeToVisitAllPoints(points))

File: leetcode/1266/code.rs
use std::cmp;

impl Solution {
    pub fn min_time_to_visit_all_points(points: Vec<Vec<i32>>) -> i32 {
        let mut time = 0;
        for i in 0..points.len()-1 {
            let x_diff = (points[i][0] - points[i+1][0]).abs();
            let y_diff = (points[i][1] - points[i+1][1]).abs();
            time += cmp::max(x_diff, y_diff);
        }
        return time
    }
}

let test_cases = vec![[[1, 1], [3, 4], [-1, 0]], [[3, 2], [-2, 2]]]
for p in 0..test_cases.len() {
    
}

